app/scripts/modules/core/src/instance/Instances.tsx

```diff
@@ -1,6 +1,133 @@
+import * as React from 'react';
+import { UIRouterContext } from '@uirouter/react-hybrid';
+import { Transition } from '@uirouter/core';
+import autoBindMethods from 'class-autobind-decorator';
+import { Subscription } from 'rxjs';
+
 import { IInstance } from 'core/domain';
 
-export interface InstancesProps {
+const equalForKeys = (keys: string[], left: any, right: any): boolean =>
+    keys.reduce((acc, key) => acc && left[key] === right[key], true);
```

Comment:
not sure the performance impact of this method (probably pretty low considering the DOM operations involved in the rendering), but might be a little faster as:
```javascript
keys.every(key => left[key] === right[key]);
```